modules/widgets/instagram/instagram.php

```diff
@@ -0,0 +1,439 @@
+	 */
+	public function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		// If coming back to the widgets page from an action, expand this widget
+		if ( isset( $_GET['instagram_widget_id'] ) && $_GET['instagram_widget_id'] == $this->number ) {
+			echo '<script type="text/javascript">jQuery(document).ready(function($){ $(\'.widget[id$="wpcom_instagram_widget-' . esc_js( $this->number ) . '"] .widget-inside\').slideDown(\'fast\'); });</script>';
+		}
+
+		// If coming back from an OAuth authentication, validate and use the one in the URL
+		if ( isset( $_GET['instagram_widget_id'] ) && $_GET['instagram_widget_id'] == $this->number
+			&& ! empty( $_GET['instagram_widget'] ) && 'connection_verified' == $_GET['instagram_widget']
+			&& ! empty( $_GET['token_id'] ) && $instance['token_id'] !== (int) $_GET['token_id'] ) {
+			//$token_store = Keyring::init()->get_token_store();
+			//$token = $token_store->get_token( array( 'type' => 'access', 'id' => (int) $_GET['token_id'] ) );
+
+			// Make sure the user owns the token ID that was passed through before using it
+			//if ( get_current_user_id() == $token->meta['user_id'] ) {
+			//	$old_token = $instance['token_id'];
+				$instance['token_id'] = (int) $_GET['token_id'];
+
+				$this->update_widget_token_id( $instance['token_id'] );
+
+			//	if ( $old_token ) {
+					//error_log( "Now delete the token: $old_token" );
+			//		$token_store->delete( array( 'type' => 'access', 'id' => $old_token ) );
+			//	}
+			//}
+		}
+		// If removing the widget's stored token ID
+		elseif ( $instance['token_id'] && isset( $_GET['instagram_widget_id'] ) && $_GET['instagram_widget_id'] == $this->number && ! empty( $_GET['instagram_widget'] ) && 'remove_token' == $_GET['instagram_widget'] ) {
+			if ( empty( $_GET['nonce'] ) || ! wp_verify_nonce( $_GET['nonce'], 'instagram-widget-remove-token-' . $this->number . '-' . $instance['token_id'] ) )
+				wp_die( __( 'Missing or invalid security nonce.', 'wpcom-instagram-widget' ) );
+
+			Keyring::init()->get_token_store()->delete( array( 'type' => 'access', 'id' => $instance['token_id'] ) );
+			$instance['token_id'] = $this->defaults['token_id'];
+
+			$this->update_widget_token_id( $instance['token_id'] );
+		}
+		// If a token ID is stored, make sure it's still valid
+		elseif ( $instance['token_id'] ) {
+			//$token = Keyring::init()->get_token_store()->get_token( array( 'type' => 'access', 'id' => $instance['token_id'] ) );
+
+			//if ( ! $token ) {
+			//	$instance['token_id'] = $this->defaults['token_id'];
+
+				$this->update_widget_token_id( $instance['token_id'] );
+			//}
+		}
+
+		// No connection, or a legacy API token? Display a connection link.
+		if ( ! $instance['token_id'] ) {
+			//$connect_url = add_query_arg( array( 'instagram_widget_id' => $this->number, 'is_customizer' => is_customize_preview() ), wpcom_keyring_get_connect_url( 'instagram_basic_display', 'instagram-widget' ) );
```

Comment:
These lines are commented out to get this working, but we'll need to use parts of them (for example adding the `is_customizer` param